.devops/deploy-pipelines.yml

```diff
@@ -326,6 +326,266 @@ stages:
-                    VALUE_FILE: "helm/pay-wallet-values-uat.yaml"
+                    VALUE_FILE: "helm/values-uat.yaml"
                     GREEN_VERSION: $(app_version)
   # --- END Deploy UAT --- #
+
+  # --- START Deploy PROD --- #
+  - stage: "Prod_WaitForApproval"
+    displayName: 'PROD approval deployment'
+    dependsOn: [ Deploy_UAT_Green,Release ]
+    condition: |
+      and(
+        eq(${{parameters.UAT_PROD_DEPLOY}}, true),
+        eq(dependencies.Deploy_UAT_Green.result, 'Succeeded'),
+        in(dependencies.Release.result, 'Succeeded', 'Skipped')
+      )
+    variables:
+      release_url: $[ stageDependencies.Release.make_release.outputs['chart_current_version.releaseUrl'] ]
+    jobs:
+      - job: Prod_Approval
+        displayName: Manual prod deploy approval
+        pool: server
+        timeoutInMinutes: 4320 # 3 days
+        steps:
+          - task: ManualValidation@0
+            timeoutInMinutes: 4320 # 3 days
+            inputs:
+              notifyUsers: $(APPROVE_TOUCHPOINT_MAIL)
+              instructions: "Please approve or reject PROD promotions for release $(release_url)"
+              onTimeout: 'reject'
+
+  - stage: "Build_PROD_Blue"
+    displayName: 'PROD blue Build'
+    dependsOn: [ Prod_WaitForApproval,Release ]
+    condition: |
+      and(
+        eq(${{parameters.UAT_PROD_DEPLOY}}, true),
+        eq(${{parameters.PROD_SKIP_BLUE_DEPLOYMENT}}, False),
+        eq(dependencies.Prod_WaitForApproval.result, 'Succeeded'),
+        in(dependencies.Release.result, 'Succeeded', 'Skipped')
+      )
+    variables:
+      app_version: $[ stageDependencies.Release.make_release.outputs['chart_current_version.appVersion'] ]
+    jobs:
+      - job: "PROD_image"
+        displayName: 'Build PROD image'
+        steps:
+          - task: Docker@2
+            displayName: "docker login for UAT"
+            inputs:
+              containerRegistry: $(UAT_CONTAINER_REGISTRY_SERVICE_CONN)
+              command: "login"
+          - task: Bash@3
+            displayName: "docker tag new version for PROD"
+            inputs:
+              targetType: "inline"
+              script: |
+                docker pull $(UAT_CONTAINER_NAMESPACE)/$(K8S_IMAGE_REPOSITORY_NAME):$(app_version)
+                docker tag $(UAT_CONTAINER_NAMESPACE)/$(K8S_IMAGE_REPOSITORY_NAME):$(app_version) $(PROD_CONTAINER_NAMESPACE)/$(K8S_IMAGE_REPOSITORY_NAME):$(Build.SourceVersion)
+          - task: Docker@2
+            displayName: "docker login for PROD"
+            inputs:
+              containerRegistry: $(PROD_CONTAINER_REGISTRY_SERVICE_CONN)
+              command: "login"
+          - task: Bash@3
+            displayName: "Push new version for PROD"
+            inputs:
+              targetType: "inline"
+              script: |
+                docker push $(PROD_CONTAINER_NAMESPACE)/$(K8S_IMAGE_REPOSITORY_NAME):$(Build.SourceVersion)
+  - stage: "Deploy_PROD_Blue"
```

Comment:
Is it useful to have blue/green deployment in prod?

Reply:
I added it mainly for homogeneity with other deployment pipelines.
IDK if we are ever going to use blue/green deployment here: maybe in the future could be useful, as for now I see it useless.

We can leave it and skip this during prod deployments (and leave it as a "disabled feature) or remove altogether and add it whenever required...
What do you think could be a better approach here?